test/system/search_test.rb

```diff
@@ -2,12 +2,86 @@
 # https://guides.rubyonrails.org/testing.html#implementing-a-system-test
 
 class SearchTest < ApplicationSystemTestCase
-  test 'searching an item from the homepage' do
-    visit '/'
+    test 'searching an item from the homepage' do
+      visit '/'
+    
+      fill_in("searchform_input", with: "Canon")
+      find('button.btn-default').click
+    
+      assert_selector('h2', text: 'Results for Canon')
+    end
+end
+
+class LoginSingUpTest < ApplicationSystemTestCase
+    test 'sign up' do
+      visit '/signup'
+    
+      assert_selector('h2', text: 'Sign up to join the Public Lab community')
+    
+      fill_in("username-signup", with: "Bob")
+      fill_in("email", with: "bob@email.com")
+    
+      # The first argument to attach_file is a locator which refers to the name
+      # attribute on the input tag.
+      # attach_file("user[photo]", Rails.root + '/public/images/pl.png')
+    
+      fill_in("password", with: "Bob1234")
+      fill_in("password-confirmation", with: "Bob1234")
+    
+      # Check the RECAPTCHA box
+      # within_frame("a-fa3pgrq3s2ty") do
+      #   check("recaptcha-anchor")
+      # end
+    
+      # find('button.btn.btn-lg.btn-primary.btn-save').click
+    
+      # This method does not require defining a custom matcher.
+      # expect(page).to have_selector(:link_or_button, 'Asking a question')
+      # expect(page).to have_selector(:link_or_button, 'Exploring projects')
+    
+      # assert_selector('h2', text: 'Hello, welcome to Public Lab!')
+    end
+end
 
-    fill_in("searchform_input", with: "Canon")
-    find('button.btn-default').click
+class QuestionTest < ApplicationSystemTestCase
+    test 'view questions by topic' do
+      visit '/questions'
+    
+      assert_selector('h4', text: 'View questions by topic')
+      fill_in("taginput", with: "sensor")
+      # expect(page).to have_content("sensor")
+    
+      find_button("View questions with the entered title").click
+      
+      # expect(page).to have_content("sensor")
+      assert_selector('h3', text: 'Notes tagged with ')
+    end
 
-    assert_selector('h2', text: 'Results for Canon')
-  end
+    test 'aske a question' do
+      visit '/questions'
```

Comment:
You've visited this URL already. All assertions should be made in the previous visit. Can you please move the assertions from there to the first one and delete everything below this point?